### ISSUE TITLE
DeadLetterPolicy, If allowAutoTopicCreation enabled in broker, the same subscriber name will conflict under different non-partitioned topics

### ISSUE THREAD

**Issue by xuesongxs** [apache/pulsar#8759]
**Describe the bug**
Consumer code:
            Consumer<String> consumer = getClient().newConsumer(Schema.STRING)
                    .topic("persistent://public/default/test-string2")
                    .subscriptionType(SubscriptionType.Shared)
                    .subscriptionName("my-subscription32")
                    .enableRetry(true)
                    .deadLetterPolicy(DeadLetterPolicy.builder()
                            .maxRedeliverCount(3)
                            .build())
                    .subscribe();

I created two non-partitioned topics: test-string2, test-string3。
The name of the subscription is the same，allowAutoTopicCreation=true，and allowAutoTopicCreationType=non-partitioned，in broker, both consumers will write messages to my-subscription32-DLQ.
The automatically created topic formart is not: topic-subscriptionName-DLQ，it is subscriptionName-DLQ.

**Comment by 315157973** [comment]
I will fix it

**Comment by 315157973** [comment]
Fixing this issue is simple. But if we modify the rules of name generation when DLQ is automatically created, there will be backward compatibility issues. So it needs to be discussed.

The current generation rules are as follows:
```Java
String deadLetterTopic = topicFirst.getNamespace() + "/" 
+ conf.getSubscriptionName() + RetryMessageUtil.DLQ_GROUP_TOPIC_SUFFIX;
```

**Comment by mattisonchao** [comment]
This issue has been fixed by #10129 , I think we should close this issue.
